prow/scripts/cluster-integration/kyma-gke-upgrade-e2e.sh

```diff
@@ -336,41 +341,52 @@ function checkTestPodTerminated() {
     return 1
 }
 
-createTestResources() {
-    shout "Create e2e upgrade test resources"
-    date
+function installTestChartOrFail() {
+  local path=$1
+  local name=$2
+  local namespace=$3
+  local label=$4
 
-    injectTestingAddons
+  shout "Create ${name} resources"
+  date
 
-    if [[  -f "$(helm home)/ca.pem" ]]; then
-        local HELM_ARGS="--tls"
-    fi
+  local HELM_ARGS
+  if [[ -f "$(helm home)/ca.pem" ]]; then
+      HELM_ARGS="--tls"
+  fi
 
-    helm install "${E2E_UPGRADE_TEST_PATH}" \
-        --name "${E2E_UPGRADE_TEST_RELEASE_NAME}" \
-        --namespace "${E2E_UPGRADE_TEST_NAMESPACE}" \
-        --timeout "${E2E_UPGRADE_TEST_HELM_TIMEOUT_SEC}" \
-        --set domain="${DOMAIN}" \
-        --wait ${HELM_ARGS}
-
-    prepareResult=$?
-    if [[ "${prepareResult}" != 0 ]]; then
-        echo "Helm install operation failed: ${prepareResult}"
-        exit "${prepareResult}"
-    fi
+  helm install "${path}" \
+      --name "${name}" \
+      --namespace "${namespace}" \
+      --timeout "${HELM_TIMEOUT_SEC}" \
+      --set domain="${DOMAIN}" \
+      --wait ${HELM_ARGS}
+
+  prepareResult=$?
+  if [[ "${prepareResult}" != 0 ]]; then
+      echo "Helm install ${name} operation failed: ${prepareResult}"
+      exit "${prepareResult}"
+  fi
 
-    set +o errexit
-    checkTestPodTerminated
-    prepareTestResult=$?
-    set -o errexit
-
-    echo "Logs for prepare data operation to test e2e upgrade: "
-    # shellcheck disable=SC2046
-    kubectl logs -n "${E2E_UPGRADE_TEST_NAMESPACE}" $(kubectl get pod -n "${E2E_UPGRADE_TEST_NAMESPACE}" -l "${E2E_UPGRADE_TEST_RESOURCE_LABEL}=${E2E_UPGRADE_TEST_LABEL_VALUE_PREPARE}" -o json | jq -r '.items | .[] | .metadata.name') -c "${TEST_CONTAINER_NAME}"
-    if [[ "${prepareTestResult}" != 0 ]]; then
-        echo "Exit status for prepare upgrade e2e tests: ${prepareTestResult}"
-        exit "${prepareTestResult}"
-    fi
+  set +o errexit
+  checkTestPodTerminated "${namespace}"
+  prepareTestResult=$?
+  set -o errexit
+
+  echo "Logs for prepare data operation to ${name}: "
+  # shellcheck disable=SC2046
+  kubectl logs -n "${namespace}" $(kubectl get pod -n "${name}" -l "${label}=${TEST_RESOURCE_LABEL_VALUE_PREPARE}" -o json | jq -r '.items | .[] | .metadata.name') -c "${TEST_CONTAINER_NAME}"
+  if [[ "${prepareTestResult}" != 0 ]]; then
+      echo "Exit status for prepare ${name}: ${prepareTestResult}"
+      exit "${prepareTestResult}"
+  fi
+}
+
+createTestResources() {
+    injectTestingAddons
+
+    installTestChartOrFail "${UPGRADE_TEST_PATH}" "${UPGRADE_TEST_RELEASE_NAME}" "${UPGRADE_TEST_NAMESPACE}" "${UPGRADE_TEST_RESOURCE_LABEL}"
+    installTestChartOrFail "${EXTERNAL_SOLUTION_TEST_PATH}" "${EXTERNAL_SOLUTION_TEST_RELEASE_NAME}" "${EXTERNAL_SOLUTION_TEST_NAMESPACE}" "${EXTERNAL_SOLUTION_TEST_RESOURCE_LABEL}"
```

Comment:
in order to shorten the runtime of this job the flow should be as follows:
install Chart1
install Chart2
waitforpodfromchart1 && output its logs
waitforpodfromchart2 && output its logs

currently, the flow is:
install chart 1 && waitforpod && output its logs
install chart 1 && waitforpod && output its logs


waiting for chart1 to completely finish is just not necessary

Reply:
Done